api/blueprints/drbCollection.py

```diff
@@ -89,6 +89,50 @@ def collectionCreate(user=None):
 
     return APIUtils.formatOPDS2Object(201, opdsFeed)
 
+@collection.route('/update/<uuid>', methods=['POST'])
+@validateToken
+def collectionUpdate(uuid, user=None):
+    logger.info('Updating current collection')
+
+    collectionData = request.json
+    dataKeys = collectionData.keys()
+
+    if len(set(dataKeys) & set(['title', 'creator', 'description'])) < 3\
+            or len(set(dataKeys) & set(['workUUIDs', 'editionIDs'])) == 0:
+        errMsg = {
+            'message':
+                'title, creator and description fields are required'
+                ', with one of workUUIDs or editionIDs to create a collection'
+        }
+
+        return APIUtils.formatResponseObject(400, 'createCollection', errMsg)
+
+    dbClient = DBClient(current_app.config['DB_CLIENT'])
+    dbClient.createSession()
+
+    #Getting the collection the user wants to replace
+    collection = dbClient.fetchSingleCollection(uuid)
```

Comment:
If a non-existent uuid is passed and we get a `sqlalchemy.orm.exc.NoResultFound` (I *think* that's what we'd expect), should we catch and return a 404? Maybe not too high pri since this is only expected to be called by an engineer running a script...

Reply:
That sounds like a good idea. It's similar to lines 104-110 in the collectionFetch endpoint. 

```
try:
        opdsFeed = constructOPDSFeed(
            uuid, dbClient, sort=sort, page=page, perPage=perPage
        )
    except NoResultFound:
        errMsg = {'message': 'Unable to locate collection {}'.format(uuid)}
        return APIUtils.formatResponseObject(404, 'fetchCollection', errMsg)
```